### PR TITLE
PA-12768 use core urls rather than only spidered collection

### PR DESCRIPTION
**Pull request by SOOS-GSteen**
### Changes:
  - use the core url count rather than the spidered count - applies to all scan types

**Ticket:** https://soos.atlassian.net/browse/PA-12768

### CODE REVIEW

File: src/zap_hooks/soos_zap_hook.py [hunk 1]
```diff
@@ -77,12 +77,16 @@ def zap_pre_shutdown(zap):
...
+    log("Spidered URLs:")
     with open('spidered_urls.txt', 'w') as f:
         for url in zap.spider.all_urls:
             f.write(f"{url}\n")
-            log(f"found: {url}")
-
+            log(f"spider discovered url: {url}")
+    log("Core URLs:")
+    with open('core_urls.txt', 'w') as f:
+        for url in zap.core.urls():
```
Comment (SOOS-GSteen) [inline src/zap_hooks/soos_zap_hook.py:87]: this is what the API scan uses to dump out the url count - https://github.com/zaproxy/zaproxy/blob/main/docker/zap-api-scan.py

File: src/constants.ts [hunk 2]
```diff
@@ -1,8 +1,9 @@
 export const SOOS_DAST_CONSTANTS = {
   AuthDelayTime: 5,
   Tool: "zap",
-  ToolVersion: "2.12",
+  ToolVersion: "2.14",
```
Comment (SOOS-GSteen) [inline src/constants.ts:4]: this should match the alpha image

File: src/zap_hooks/soos_zap_hook.py (hunk 1 above)
Comment (SOOS-GSteen) [reply src/zap_hooks/soos_zap_hook.py:87]: logging.info('Number of Imported URLs: ' + str(len(urls)))

and

print('Total of ' + str(num_urls) + ' URLs')

(which are different before and after scan

File: src/index.ts [hunk 3]
```diff
@@ -364,8 +364,12 @@ class SOOSDASTAnalysis {
       });
       soosLogger.info(`Scan result uploaded successfully`);
 
+
+      if (data["coreUrls"]?.length) {
```
Comment (SOOS-GSteen) [inline src/index.ts:368]: image

Comment (SOOS-GSteen) [reply src/index.ts:368]: going to just use the core url count rather than the spider count - and leave the json element the same

**Review by SOOS-MMalony — APPROVED** [review]

**Pull request merged**